packages/block-editor/src/utils/object.js

```diff
@@ -125,7 +128,7 @@ export function setImmutably( object, path, value ) {
  * @return {*} Value of the object property at the specified path.
  */
 export const getValueFromObjectPath = ( object, path, defaultValue ) => {
-	const normalizedPath = Array.isArray( path ) ? path : path.split( '.' );
+	const normalizedPath = Array.isArray( path ) ? path : stringToPath( path );
```

Comment:
This can totally be shipped as a separate PR regardless of how we decide to move forward.